frontend/src/views/consumer/paymentstatus-modal/index.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+import Modal, { Content } from '../../../components/modal';
+import Typography from '../../../components/typography';
+import EventCard from '../../../components/event-card';
+import { connect, parseQuery } from '../../../utils';
+import styled, { withTheme } from 'styled-components';
+import { Route } from 'react-router-dom';
+import chroma from 'chroma-js';
+
+const EventCardWrapper = styled.div`
+  background-color: ${props =>
+    chroma(props.bgColor)
+      .alpha(0.2)
+      .css('rgba')};
+  padding: 2rem;
+  border-radius: 0 0 8px 8px;
+`;
+
+export default withTheme(
+  connect('store')(
+    class PaymentStatusModal extends React.Component {
+      state = {
+        show: true,
+      };
+      getColor = statusCode => {
+        const { red, primary, lightGreen, purple } = this.props.theme.palette;
+        if (statusCode == 0) return lightGreen;
```

Comment:
Consider using switch case here or creating a mapping table for these statuses, eg.
```
const STATUSES = {
  0: {
    color: lightGreen,
    meaning: 'Maksu vahvistettu',
    error: "Kiitos ostostasi! Lähetämme liput antamaasi puhelinnumeroon tekstiviestillä"
  },
  1: {
    color: red,
    meaning: 'Payment not in system',
    error: "It seems like you have stumbled upon a weird error. Please let our engineer know what happened"
  },
  2: {
    color: primary,
    meaning: 'Already paid',
    error: "You already paid for this event. The order that you requested are therefore not fulfilled. You have not been charge anything."
  },
  ...
};
```
Then you could just do eg. `const meaning = STATUSES[statusCode]['meaning'];`

Reply:
You could even do enumeration for status codes to make the code more readable.

Reply:
i think its just the approach. Getter function works just fine for me.